modules/transports/core/nhttp/src/main/java/org/apache/synapse/transport/passthru/TargetResponse.java

```diff
@@ -206,4 +219,18 @@ public NHttpClientConnection getConnection() {
     public ProtocolVersion getVersion() {
         return version;
     }
+
+    /**
+     * Retrieve the keep-alive timeout value
+     * @param keepAlive header value
+     * @return keep-alive timeout
+     */
+    private String getKeepAliveTimeout(String keepAlive) {
+        String timeout = keepAlive.split("timeout=")[1];
```

Comment:
This might throw array index out of bound exception if we don't have **=** sign right after the timeout header.